x/accesscontrol/types/access_operations.go

```diff
@@ -0,0 +1,74 @@
+package types
+
+import (
+	acltypes "github.com/cosmos/cosmos-sdk/types/accesscontrol"
+)
+
+type AccessOperationSet struct {
+	ops map[acltypes.AccessOperation]struct{}
+}
+
+func NewEmptyAccessOperationSet() *AccessOperationSet {
+	return &AccessOperationSet{ops: map[acltypes.AccessOperation]struct{}{}}
+}
+
+func NewAccessOperationSet(ops []acltypes.AccessOperation) *AccessOperationSet {
+	set := NewEmptyAccessOperationSet()
+	set.AddMultiple(ops)
+	return set
+}
+
+func (waos *AccessOperationSet) Add(op acltypes.AccessOperation) {
+	waos.ops[op] = struct{}{}
+}
+
+func (waos *AccessOperationSet) AddMultiple(ops []acltypes.AccessOperation) {
+	for _, op := range ops {
+		waos.Add(op)
+	}
+}
+
+func (waos *AccessOperationSet) Merge(other *AccessOperationSet) {
+	for op := range other.ops {
+		waos.Add(op)
+	}
+}
+
+func (waos *AccessOperationSet) Has(op acltypes.AccessOperation) bool {
+	_, ok := waos.ops[op]
+	return ok
+}
+
+func (waos *AccessOperationSet) Size() int {
+	return len(waos.ops)
+}
+
+func (waos *AccessOperationSet) ToSlice() []acltypes.AccessOperation {
+	res := []acltypes.AccessOperation{}
+	for op := range waos.ops {
+		if op != *CommitAccessOp() {
+			res = append(res, op)
+		}
+	}
+	res = append(res, *CommitAccessOp())
```

Comment:
it's not guaranteed to have a commit access op right? its probably fine to always append but is there any scenario where we would intend to not append the commit IF it doesnt exist in the map?

Reply:
changed so that it only appends if it exists